### PR TITLE
Add like button and save conversation

### PR DESCRIPTION
**Pull request by alalulu8668**
add like button to collect feedback, store each session (with user feedback) as json file.

### CODE REVIEW

File: bioimageio_chatbot/chatbot.py [hunk 1]
```diff
@@ -181,8 +183,24 @@ async def respond_to_user(question_with_history: QuestionWithHistory = None, rol
...
+
+    # Create a chat_log.json file inside the session folder
+    chat_log_path = os.path.join("chat_sessions", f"{session_id}-chatlogs.json")
+
+    chat_his_dict = {'conversations':chat_history, 'timestamp': str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")), 'user': context['user']}
+    # Serialize the chat history to a json string
+    chat_history_json = json.dumps(chat_his_dict)
+
+    # Write the serialized chat history to the json file
+    with open(chat_log_path, 'w', encoding='utf-8') as file:
```
Comment (oeway) [inline bioimageio_chatbot/chatbot.py:198]: Would be better if we use aiofiles to write it: https://github.com/Tinche/aiofiles

File: bioimageio_chatbot/chatbot.py [hunk 2]
```diff
@@ -199,7 +217,10 @@ async def register_chat_service(server):
     description_by_id = {collection['id']: collection['description'] for collection in collections}
     customer_service = create_customer_service(knowledge_base_path)
 
-    async def chat(text, chat_history, user_profile=None, channel=None, status_callback=None, context=None):
+    async def report(user_report, context=None):
+        print(f"UserFeedback: {user_report}")
```
Comment (oeway) [inline bioimageio_chatbot/chatbot.py:221]: We need to call the save chat history here too. The difference is that it should contain additional key "comment" and "like", and save in a different folder.

**Review by oeway — CHANGES_REQUESTED** [review]

File: bioimageio_chatbot/static/index.html [hunk 3]
```diff
@@ -246,6 +256,16 @@ <h2 class="text-center">Welcome to BioImage.IO Chatbot</h2>
           </button>
         </div>
 
+        <!-- Feedback form initially collapsed -->
+        <div id="feedback-form" class="collapse">
+          <textarea class="form-control mt-3" id="generalFeedback" placeholder="Tell us your experience about Bioimage.IO chatbot..." rows="3"></textarea>
+          <!-- Save Feedback button -->
+          <button type="button" class="btn btn-success mt-3" id="save-feedback-btn">
+            Send Feedback
```
Comment (oeway) [inline bioimageio_chatbot/static/index.html:264]: Change to `Submit`

File: bioimageio_chatbot/static/index.html [hunk 4]
```diff
@@ -653,7 +667,46 @@ <h3><div class="spinner" style='display: inline-block;width:25px;height:25px;mar
...
+    });
+
+    // Add a click event handler for the "Save Feedback" button
+    $('#save-feedback-btn').click(function () {
+      // Get the general feedback message from the textarea
+      const generalFeedback = $('#generalFeedback').val();
+
+      // Create a feedbackData object
+      const feedbackData = {
+        type: 'general feedback', // Set the type to 'general feedback'
```
Comment (oeway) [inline bioimageio_chatbot/static/index.html:690]: Maybe change to `type: 'general'`

File: bioimageio_chatbot/chatbot.py [hunk 5]
```diff
@@ -275,7 +317,7 @@ async def index(event, context=None):
 
 if __name__ == "__main__":
     # asyncio.run(main())
-    server_url = "https://ai.imjoy.io"
+    server_url = "https://chat.bioimage.io"
```
Comment (oeway) [inline bioimageio_chatbot/chatbot.py:320]: Change back to `https://ai.imjoy.io`

File: requirements.txt [hunk 6]
```diff
@@ -3,4 +3,5 @@ imjoy-rpc
 requests
 pypdf
 hypha
-tqdm
\ No newline at end of file
+tqdm
+aiofiles
```
Comment (oeway) [inline requirements.txt:7]: Add a new line

**Comment by alalulu8668** [comment]
All updates according to Wei's comments.

**Review by oeway — APPROVED** [review]
Thanks! Looks good to me now!

**Pull request merged**